src/store/workbenchReducer.ts

```diff
@@ -1,14 +1,26 @@
 import { PayloadAction } from '@reduxjs/toolkit';
 import { IColumnDesc } from 'lineupjs';
 import { IRow } from 'tdp_core';
-import { EWorkbenchDirection, ISelectedMapping, IWorkbench } from './interfaces';
+import { EWorkbenchDirection, ISelectedMapping, IWorkbench, IOrdinoAppState } from './interfaces';
 
 export const workbenchReducers = {
-  addFirstWorkbench(state, action: PayloadAction<IWorkbench>) {
+  addFirstWorkbench(
+    state,
+    action: PayloadAction<{
+      workbench: IWorkbench;
+      globalQuery: IOrdinoAppState['globalQuery'];
+      appliedQueryFilter: IOrdinoAppState['appliedQueryFilter'];
```

Comment:
```suggestion
      globalQuery: IOrdinoGlobalQuery;
      appliedQueryFilter: IQueryFilter;
```

I assume this can then be replaced, too.